File: 66.plus-one.js
/*
 * @lc app=leetcode id=66 lang=javascript
 *
 * [66] Plus One
 */

// @lc code=start
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function (digits) {
  for (let i = digits.length - 1; i >= 0; --i) {
    if (digits[i] === 9) {
      digits[i] = 0;
    } else {
      ++digits[i];
      return digits;
    }
  }
  digits[0] = 1;
  digits.push(0);
  return digits;
};
// @lc code=end

File: 66.plus-one.go
/*
 * @lc app=leetcode id=66 lang=golang
 *
 * [66] Plus One
 */

package leetcode

// @lc code=start
func plusOne(digits []int) []int {
	for i := len(digits) - 1; i > -1; i-- {
		if digits[i] < 9 {
			digits[i]++
			return digits
		}
		digits[i] = 0
	}
	digits[0] = 1
	digits = append(digits, 0)
	return digits
}

// @lc code=end
